tchannel/tornado/connection.py

```diff
@@ -497,6 +512,16 @@ def ping(self):
     def pong(self):
         return self._write(messages.PingResponseMessage())
 
+    def add_incoming_request(self, request):
+        self.incoming_requests[request.id] = request
+
+    def get_incoming_request(self, id):
+        return self.incoming_requests.get(id, None)
+
+    def remove_incoming_request(self, id):
+        req = self.incoming_requests.pop(id, None)
+        return req
```

Comment:
Add docs for all three methods that provide more context. What are the arguments? When/why is this method called?